scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncDoFn.java

```diff
@@ -64,9 +61,10 @@ public void finishBundle(FinishBundleContext c) {
         waitForFutures(futures.values());
       } catch (InterruptedException e) {
         Thread.currentThread().interrupt();
+        LOG.error("Failed to process futures", e);
         throw new RuntimeException("Failed to process futures", e);
       } catch (ExecutionException e) {
-        throw new RuntimeException("Failed to process futures", e);
+        LOG.error("Failed to process futures", e);
```

Comment:
Why take away the RuntimeException? Shouldn't the bundle get failed ?

Reply:
You're right.
I had to remove it for the `BaseAsyncLookupDoFn` case since we don't want to fail a bundle if there're lookup errors. In that case the callback appends into the error queue and we surface it to user as a failed `Try`.
In this case we should rethrow.